manager/dispatcher/dispatcher.go

```diff
@@ -216,6 +217,9 @@ func (d *Dispatcher) Run(ctx context.Context) error {
 
 	defer cancel()
 	d.ctx, d.cancel = context.WithCancel(ctx)
+	ctx = d.ctx
```

Comment:
Can I assume context assignment as derived context, i.e., when `d.cancel()` is called, `ctx` from `ctx = d.ctx` is also cancelled?